src/miner.cpp

```diff
@@ -331,18 +331,52 @@ CMutableTransaction CreateCoinbaseTransaction(const CChainParams& chainparams, C
         return mtx;
 }
 
-CBlockTemplate* CreateNewBlock(const CChainParams& chainparams, const MinerAddress& minerAddress, const std::optional<CMutableTransaction>& next_cb_mtx)
+BlockAssembler::BlockAssembler(const CChainParams& _chainparams)
+    : chainparams(_chainparams)
 {
-    // Create new block
-    std::unique_ptr<CBlockTemplate> pblocktemplate(new CBlockTemplate());
+   // Largest block you're willing to create:
+    nBlockMaxSize = GetArg("-blockmaxsize", DEFAULT_BLOCK_MAX_SIZE);
+    // Limit to between 1K and MAX_BLOCK_SIZE-1K for sanity:
+    nBlockMaxSize = std::max((unsigned int)1000, std::min((unsigned int)(MAX_BLOCK_SIZE-1000), nBlockMaxSize));
```

Comment:
If this is intended to reserve space for the coinbase transaction, 1000 bytes may not be large enough. Nonblocking for this PR but it should be addressed before or as part of the ZIP 317 changes.